### ISSUE TITLE
Analytics form has no labels for accessibility

### ISSUE THREAD

**Issue by steverep** [home-assistant/frontend#11463]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

The form to turn on or off sharing of analytics has no labels or descriptions associated with the various checkboxes, making them poorly accessible to assistive technology.

### Describe the behavior you expected

1. A concise label should be associated with each checkbox using `<label>`, `aria-label`, or `aria-labeledby`.
2. The detailed text for each checkbox should be referenced using `aria-describedby`.

### Steps to reproduce the issue

1.
2.
3.
...
1. Go to the analytics form either during onboarding or via configuration
2. Turn on any screen reader (e.g. NVDA) and tab through the form

### What version of Home Assistant Core has the issue?

2021.12.10

**Comment by steverep** [comment]
As I'm paying more attention now, I'm realizing this is not just the analytics form.  There are lots of checkboxes that need fixing.  For example, all checkboxes on a user profile page are unlabelled.

**Comment by steverep** [comment]
Also found a pair of malformed labels in [dialog-config-entry-system-options](https://github.com/home-assistant/frontend/blob/4e0cebaf32f042377021536289a73b354b921145/src/dialogs/config-entry-system-options/dialog-config-entry-system-options.ts), probably because HTML elements are being shoved directly into the label attribute.  NVDA announces the label as "[object Object]".

**Comment by steverep** [comment]
Still an issue of course...

**Comment by steverep** [comment]
Still an issue

**Comment by steverep** [comment]
Still an issue

**Comment by rekinet** [comment]
Unfortunately, the problem still persists. There are missing labels for checkbox switches in HA: analytics data, backup list, entity list, profile options, for network adapter options, and for configuration options for NabuCasa: remote access, Alexa, state reporting, Google Assistant, and also state reporting.

**Comment by Nardol** [comment]
Still an issue today.

**Comment by Nardol** [comment]
I still observe this behavior with Home Assistant 2024.2.2.
